src/Icons/tests/Unit/IconRendererTest.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Symfony\UX\Icons\Tests\Unit;
+
+use InvalidArgumentException;
+use PHPUnit\Framework\TestCase;
+use Symfony\UX\Icons\Exception\IconNotFoundException;
+use Symfony\UX\Icons\IconRegistryInterface;
+use Symfony\UX\Icons\IconRenderer;
+use Symfony\UX\Icons\Registry\InMemoryIconRegistry;
+use Symfony\UX\Icons\Svg\Icon;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ */
+class IconRendererTest extends TestCase
+{
+    public function testRenderIcon(): void
+    {
+        $registry = $this->createRegistry([
+            'user' => '<circle ',
+        ]);
+        $iconRenderer = new IconRenderer($registry);
+
+        $icon = $iconRenderer->renderIcon('user');
+
+        $this->assertStringContainsString('<svg', $icon);
+        $this->assertStringContainsString('<circle', $icon);
+    }
+
+    public function testRenderIconThrowsExceptionWhenIconNotFound(): void
+    {
+        $registry = $this->createRegistry([]);
+        $iconRenderer = new IconRenderer($registry);
+
+        $this->expectException(IconNotFoundException::class);
+
+        $iconRenderer->renderIcon('foo');
+    }
+
+    public function testRenderIconThrowsExceptionWhenAttributesAreInvalid(): void
+    {
+        $registry = $this->createRegistry(['foo' => '<path d="M0 0L12 12"/>']);
+        $iconRenderer = new IconRenderer($registry);
+
+        $this->expectException(InvalidArgumentException::class);
```

Comment:
Exception message should also be tested, no?

Reply:
No i just want to test exceptions are thrown from Icon. 

The message is not defined there so won't test it.